src/controllers/controllers/controller.py

```diff
@@ -153,33 +179,80 @@ def get_observation(self, policy):
 
         odom, lidar = self.get_data()
         odom = process_odom(odom)
-        if self.firstOdom:
-            self.offset = odom[0:6]
-            self.firstOdom = False
-        odom[0] = odom[0] - self.offset[0]
-        odom[1] = odom[1] - self.offset[1]
-        odom[2] = odom[2] - self.offset[2]
-        odom[3] = odom[3] - self.offset[3]
-        odom[4] = odom[4] - self.offset[4]
-        odom[5] = odom[5] - self.offset[5]
+
+        limited_odom = odom[-2:]
+        # if self.firstOdom:
+        #     self.offset = odom[0:6]
+        #     self.firstOdom = False
+        # odom[0] = odom[0] - self.offset[0]
+        # odom[1] = odom[1] - self.offset[1]
+        # odom[2] = odom[2] - self.offset[2]
+        # odom[3] = odom[3] - self.offset[3]
+        # odom[4] = odom[4] - self.offset[4]
+        # odom[5] = odom[5] - self.offset[5]
+        
         num_points = self.LIDAR_POINTS
 
-        if policy == 'ftg':
-            lidar_range = forward_reduce_lidar(lidar)
-        else:
-            lidar_range = avg_lidar(lidar, num_points)
 
-        # TODO: find out how to deal with this better. 
-        # Testing code for pre trained AE
-        # # ae_range = process_ae_lidar_beta_vae(lidar, self.ae_lidar_model, is_latent_only=False) #[1, 1, 512]
-        # ae_range = process_ae_lidar(lidar, self.ae_lidar_model,is_latent_only = False)
-        # # scan = create_lidar_msg(lidar, num_points, lidar_range)
-        # scan = create_lidar_msg(lidar, len(ae_range), ae_range)
+        match self.LIDAR_PROCESSING:
+            case 'avg':
+                processed_lidar_range = avg_lidar(lidar, num_points)
+                visualized_range = processed_lidar_range
+                scan = create_lidar_msg(lidar, num_points, visualized_range)
+            case 'raw':
+                processed_lidar_range = process_lidar_med_filt(lidar, 15)
+                visualized_range = processed_lidar_range
+                scan = create_lidar_msg(lidar, num_points, visualized_range)
+            case 'avg_w_consensus':
+                processed_lidar_range = avg_lidar_w_consensus(lidar, num_points)
+                visualized_range = processed_lidar_range
+                scan = create_lidar_msg(lidar, num_points, visualized_range)
+            case 'forward_reduce':
+                processed_lidar_range = forward_reduce_lidar(lidar)
+
 
-        scan = create_lidar_msg(lidar, len(lidar_range), lidar_range)
         self.processed_publisher.publish(scan)
```

Comment:
Not all cases seem to produce a variable called scan so this could break the code.